pkg/virt-controller/services/rendercontainer.go

```diff
@@ -169,7 +169,7 @@ func securityContext(userId int64, privileged bool, requiredCapabilities *k8sv1.
 }
 
 func getPortsFromVMI(vmi *v1.VirtualMachineInstance) []k8sv1.ContainerPort {
-	ports := make([]k8sv1.ContainerPort, 0)
+	var ports []k8sv1.ContainerPort
```

Comment:
Can you define ports at the return argument? In this way this variable isn't needed

Reply:
I agree with @alicefr :+1:

Reply:
Do you mean using a named return argument ? If so, I'm sorry, but while I can, I won't :) I think that entire feature is mistake. 

I think this makes it more readable, and explicit.

Reply:
Well, I think it's usually a mistake to say something like: "it's NEVER the right thing to do under any circumstance". I think that there are problems with this feature, but sometimes it fits perfectly.

But we can agree to disagree.

Reply:
Yeah, I have to eat my own words. It's always a mistake to say something that polarizing. Sorry for how I've expressed myself.

In this circumstance though, I fail to see the advantage of using a named return variable - on the other hand, I think it is more explicit - thus easier to follow and reason about - if we just declared a variable, append to it when appropriate, and return it.

Reply:
@maiqueb I understand your point, tbh it isn't my favorite either. However, we use it in KubeVirt code. So, I think it will be better to decide on a single way and be coherent in the code. Additionally, certain go linters they suggest to use it in this way.

Reply:
@maiqueb not polarizing at all, every opinion is welcome here :) no need to apologize.

I think that in this case the advantage is to save a line of code for initialization. I don't see how it makes anything more explicit, anyone familiar with Go knows that feature exists and expects variables to be defined this way. I agree it's not anywhere near critical, but just my opinion :) but again we can agree to disagree.

@alicefr I agree with the general approach, but I think this has to be discussed in depth before making such decisions, especially where we want to ban a feature that is a core language feature that is broadly accepted among Go developers. TBH, for such decisions, I think it might be better to have different styles then enforcing one single style, especially when it's a non-idiomatic way of using the language. We should start with the obvious stuff first :)

Reply:
fine for me :)

Reply:
I agree with @maiqueb , especially in this case.
Named return values are discouraged and should be used only if there is no other choice.
(The common case is when defer is used and the error needs to be evaluated or changed by it)

You can [explore](https://dave.cheney.net/practical-go/presentations/gophercon-israel.html#_avoid_named_return_values) more experienced engineers who claim the same.
My main part that convinced me is:
> named return values are a symptom of a clever piece of code which should be reviewed with suspicion.

If I see it in the code, I immediately start looking for the "catch".

Please, pretty please, do not use it to make writing the code shorter, prioritize for clarity to the next reader.